File: app/src/test/java/com/farmerbb/taskbar/ui/TestUIController.java
package com.farmerbb.taskbar.ui;

import android.content.Context;

public class TestUIController extends UIController {
    UIHost onCreateHost;
    UIHost onRecreateHost;
    UIHost onDestroyHost;

    public TestUIController(Context context) {
        super(context);
    }

    @Override
    void onCreateHost(UIHost host) {
        onCreateHost = host;
    }

    @Override
    void onRecreateHost(UIHost host) {
        onRecreateHost = host;
    }

    @Override
    void onDestroyHost(UIHost host) {
        onDestroyHost = host;
    }
}

File: app/src/test/java/com/farmerbb/taskbar/ui/TestUIController.kt
package com.farmerbb.taskbar.ui

import android.content.Context

class TestUIController(context: Context?) : UIController(context) {
    var onCreateHost: UIHost? = null
    var onRecreateHost: UIHost? = null
    var onDestroyHost: UIHost? = null
    public override fun onCreateHost(host: UIHost) {
        onCreateHost = host
    }

    public override fun onRecreateHost(host: UIHost) {
        onRecreateHost = host
    }

    public override fun onDestroyHost(host: UIHost) {
        onDestroyHost = host
    }
}
